lib/perl/Genome/Qc/Tool/WithVariationListVcf.pm

```diff
@@ -0,0 +1,37 @@
+package Genome::Qc::Tool::WithVariationListVcf;
+
+use strict;
+use warnings;
+use Genome;
+
+class Genome::Qc::Tool::WithVariationListVcf {
+    is_abstract => 1,
+    has => [
+        variation_list_build_id => {
+            is => 'Text',
+            is_optional => 1,
+        },
+        variation_list_build => {
+            is => 'Genome::Model::Build::ImportedVariationList',
+            id_by => 'variation_list_build_id',
+        },
+    ],
+};
+
+sub variation_list_vcf_file {
+    my $self = shift;
+
+    my $vcf_file = Genome::Sys->create_temp_file_path;
+    my $cmd = Genome::Model::GenotypeMicroarray::Command::ExtractToVcf->create(
+        variation_list_build => $self->variation_list_build,
+        sample => $self->sample,
```

Comment:
It's a subroutine on `Genome::Qc::Tool`. `Genome::Qc::Tool::WithVariationListVcf` is supposed to work as an interface that qc tools can inherit from, in addition to inheriting from `Genome::Qc::Tool`. `Genome::Qc::Tool::WithVariationListVcf` is supposed to provide additional functionality for certain qc tool child classes. I guess I could make  `Genome::Qc::Tool::WithVariationListVcf` inherit from `Genome::Qc::Tool` and have the appropriate tools (like `Genome::Qc::Tool::Bedtools::Intersect`) inherit only from `Genome::Qc::Tool::WithVariationListVcf` and not from both `[Genome::Qc::Tool, Genome::Qc::Tool::WithVariationListVcf]`. Do you prefer that inheritance hierarchy?